src/lib/snykGroup.ts

```diff
@@ -69,6 +80,29 @@ export class snykGroup {
     } catch (err: any) {
       utils.log(err);
     }
+
+    //get roles
+    try{
+      let response = await this._requestManager.request({
+        verb: 'GET',
+        url:`/group/${this.id}/roles`
+      });
+      //map roles to role id
+      response.data.map( (currRole: any) => this._roles[currRole["name"].toUpperCase()] = currRole["publicId"]);
```

Comment:
something doesn't feel right about mapping the list results the API already gives us into a dictionary with key value/pairs and changing the mental model of the roles from a list dictionary objects to a single dictionary object.

can we keep the roles response data as is and map it to `this._roles` as a list (this is then consistent with how we are already populating `this._members` and `this._orgs`, etc...

let's also add an interface definition in `types.ts` for `GroupRole`, so we can set the type for `this._roles = GroupRole[]`

then we can put the advanced logic for dealing with the default vs custom roles with same name in another private function in this class, and the mapping can happen inside there? I think it will be easier to follow WDYT?

Reply:
Yeah I agree, that's much easier to follow, I'll start making those changes.

Reply:
Made the requested changes, let me know what you think